### ISSUE TITLE
Spring-Boot 1.4.2 Compatibility

### ISSUE THREAD

**Issue by jvillane** [pac4j/spring-security-pac4j-boot-demo#1]
Maybe this isn't an ISSUE, but i tried to make this project work with spring-boot 1.4.2 and it doesn't start

Is there any intention of making a newer version of the project?

Thanks in advance (from Chile)!

**Comment by jvillane** [comment]
I founded a solution! After changing the spring-boot version:

    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>1.4.2.RELEASE</version>
    </parent>

You need to include the dependency:

    <dependency>
        <groupId>joda-time</groupId>
        <artifactId>joda-time</artifactId>
        <version>2.9.4</version>
    </dependency>

And changing the Pac4jConfig class as follows...

    @Configuration
    @Component
    public class Pac4jConfig extends Config {

        @Value("${salt}")
        private String salt;

        public Pac4jConfig() {
            final OidcConfiguration oidcConfiguration = new OidcConfiguration();
            oidcConfiguration.setClientId("167480702619-8e1lo80dnu8bpk3k0lvvj27noin97vu9.apps.googleusercontent.com");
            oidcConfiguration.setSecret("MhMme_Ik6IH2JMnAT6MFIfee");
            oidcConfiguration.setPreferredJwsAlgorithm(JWSAlgorithm.PS384);
            oidcConfiguration.addCustomParam("prompt", "consent");
            final GoogleOidcClient oidcClient = new GoogleOidcClient(oidcConfiguration);
            oidcClient.setAuthorizationGenerator(profile -> profile.addRole("ROLE_ADMIN"));

            final SAML2ClientConfiguration cfg = new SAML2ClientConfiguration("resource:samlKeystore.jks",
                "pac4j-demo-passwd",
                "pac4j-demo-passwd",
                "resource:metadata-okta.xml");
            cfg.setMaximumAuthenticationLifetime(3600);
            cfg.setServiceProviderEntityId("http://localhost:8080/callback?client_name=SAML2Client");
            cfg.setServiceProviderMetadataPath("sp-metadata.xml");
            final SAML2Client saml2Client = new SAML2Client(cfg);

            final FacebookClient facebookClient = new FacebookClient("145278422258960", "be21409ba8f39b5dae2a7de525484da8");
            final TwitterClient twitterClient = new TwitterClient("CoxUiYwQOSFDReZYdjigBA",
                "2kAzunH5Btc4gRSaMr7D7MkyoJ5u1VzbOOzE8rBofs");
            // HTTP
            final FormClient formClient = new FormClient("http://localhost:8080/loginForm.jsp", new SimpleTestUsernamePasswordAuthenticator());
            final IndirectBasicAuthClient indirectBasicAuthClient = new IndirectBasicAuthClient(new SimpleTestUsernamePasswordAuthenticator());

            // CAS
            final CasClient casClient = new CasClient("https://casserverpac4j.herokuapp.com/login");
            // casClient.setGateway(true);

            // REST authent with JWT for a token passed in the url as the token parameter
            ParameterClient parameterClient = new ParameterClient("token", new JwtAuthenticator(salt));
            parameterClient.setSupportGetRequest(true);
            parameterClient.setSupportPostRequest(false);

            // basic auth
            final DirectBasicAuthClient directBasicAuthClient = new DirectBasicAuthClient(new SimpleTestUsernamePasswordAuthenticator());

            final AnonymousClient anonymousClient = new AnonymousClient();

            final Clients clients = new Clients("http://localhost:8080/callback", oidcClient, saml2Client, facebookClient,
                twitterClient, formClient, indirectBasicAuthClient, casClient, parameterClient, directBasicAuthClient, anonymousClient);

            this.setClients(clients);
            this.addAuthorizer("admin", new RequireAnyRoleAuthorizer("ROLE_ADMIN"));
            this.addAuthorizer("custom", new CustomAuthorizer());
            this.addMatcher("excludedPath", new ExcludedPathMatcher("^/facebook/notprotected\\.html$"));
        }
    }

And it works!!

**Comment by leleuj** [comment]
Indeed, it is meant to work with the latest version of Spring Boot. Apart from the Joda dependency, what did you change in the `Pac4jConfig`?

**Comment by jvillane** [comment]
I detailed the changes in the Pac4jConfig class above (in the previous comment) but i didn't mentioned that i had to delete the **pac4j-mongo** dependency

PD: @leleuj thank you very much for your working project!
